src/Interpreters/Cache/FileCacheSettings.cpp

```diff
@@ -1,174 +1,207 @@
+            else if (type_name == "String")
+                return std::make_shared<DataTypeString>();
+            else if (type_name == "Bool")
+                return std::make_shared<DataTypeUInt8>();
+            else if (type_name == "Double")
+                return std::make_shared<DataTypeFloat64>();
+            else
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unexpected type: {}", type_name);
+        }();
+        desc.comment = setting.getDescription();
+        result.add(desc);
     }
 
-    if (has("slru_size_ratio"))
-        slru_size_ratio = get_double("slru_size_ratio");
+    result.add(
+        ColumnDescription(
+            "is_initialized", std::make_shared<DataTypeUInt8>(), "Indicates whether cache was successfully initialized"));
 
-    if (has("write_cache_per_user_id_directory"))
-        slru_size_ratio = get_uint("write_cache_per_user_id_directory");
+    result.add(
+        ColumnDescription(
+            "current_size", std::make_shared<DataTypeUInt64>(), "Current cache size"));
+    result.add(
+        ColumnDescription(
+            "current_elements_num", std::make_shared<DataTypeUInt64>(), "Current cache elements (file segments) number"));
 
-    if (has("keep_free_space_size_ratio"))
-        keep_free_space_size_ratio = get_double("keep_free_space_size_ratio");
+    return result;
+}
 
-    if (has("keep_free_space_elements_ratio"))
-        keep_free_space_elements_ratio = get_double("keep_free_space_elements_ratio");
+void FileCacheSettings::dumpToSystemSettingsColumns(
+    MutableColumnsAndConstraints & params,
+    const std::string & cache_name,
+    const FileCachePtr & cache) const
+{
+    MutableColumns & res_columns = params.res_columns;
+    size_t i = 0;
+    res_columns[i++]->insert(cache_name);
 
-    if (has("keep_free_space_remove_batch"))
-        keep_free_space_elements_ratio = get_uint("keep_free_space_remove_batch");
+    for (const auto & setting : impl->all())
+        res_columns[i++]->insert(setting.getValue());
+
+    res_columns[i++]->insert(cache->isInitialized());
+    res_columns[i++]->insert(cache->getUsedCacheSize());
+    res_columns[i++]->insert(cache->getFileSegmentsNum());
 }
 
 void FileCacheSettings::loadFromConfig(const Poco::Util::AbstractConfiguration & config, const std::string & config_prefix)
 {
-    auto config_has = [&](std::string_view key) { return config.has(fmt::format("{}.{}", config_prefix, key)); };
-    auto config_get_uint = [&](std::string_view key) { return config.getUInt(fmt::format("{}.{}", config_prefix, key)); };
-    auto config_get_string = [&](std::string_view key) { return config.getString(fmt::format("{}.{}", config_prefix, key)); };
-    auto config_get_double = [&](std::string_view key) { return config.getDouble(fmt::format("{}.{}", config_prefix, key)); };
-    loadImpl(std::move(config_has), std::move(config_get_uint), std::move(config_get_string), std::move(config_get_double));
+    if (!config.has(config_prefix))
```

Comment:
Is this a bug? The required semantics here should be `hasPrefix`, but the semantics of `config.has()` are `hasKey`

Reply:
@kssenii

Reply:
There is no method in `Poco::Util::AbstractConfiguration` as `hasPrefix`, there is only `has` and it must work with prefixes.